netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,251 @@
+  initUI()
+
+  def syncTheme(): Unit = {
+    setBackground(InterfaceColors.MONITOR_BACKGROUND)
+    messageLabel.setForeground(InterfaceColors.DISPLAY_AREA_TEXT)
+    closeButton.setForeground(InterfaceColors.DISPLAY_AREA_TEXT)
+  }
+
+  // Method to fetch JSON content from a URL
+  def fetchJsonFromUrl(url: String): String = {
+    try {
+      val source = Source.fromURL(url)
+      val content = source.mkString
+      source.close()
+      content
+    } catch {
+      case e: Exception =>
+      println(s"Unable to connect to ccl.northwestern.edu but not throwing an exception")
+      ""
+//        throw new Exception(s"Error fetching JSON from URL: ${e.getMessage}", e)
+    }
+  }
+
+  private def initUI(): Unit = {
+    // Set layout and appearance
+    setLayout(new BorderLayout())
+    setPreferredSize(new Dimension(400, 50))
+
+    // Customize the message label
+    messageLabel.setHorizontalAlignment(SwingConstants.LEFT)
+    add(messageLabel, BorderLayout.CENTER)
+
+    // Configure the close button
+    closeButton.setBorderPainted(false)
+    closeButton.setContentAreaFilled(false)
+    closeButton.setFocusPainted(false)
+    //TODO this action needs to get called on the okay button click.
+    closeButton.addActionListener(new ActionListener {
+      override def actionPerformed(e: ActionEvent): Unit = {
+        setVisible(false) // Hide the banner when the close button is pressed
+      }
+    })
+    add(closeButton, BorderLayout.EAST)
+
+    // Add a mouse listener to call showJsonInDialog when the banner is clicked
+    addMouseListener(new MouseAdapter {
+      override def mouseClicked(e: MouseEvent): Unit = {
+        showJsonInDialog() // Call the method on click
+      }
+    })
+  }
+
+  // Method to update the notification message
+  def setMessage(message: String): Unit = {
+    messageLabel.setText(" " + message)
+    setVisible(true) // Ensure the banner is visible when updating the message
+  }
+
+  // Method to parse JSON content to a list of JsonObject instances
+  def parseJsonToList(jsonContent: String): List[JsonObject] = {
+    if(jsonContent.isEmpty){
+      return null
+    }
+
+    try {
+      val parser = new JSONParser()
+      val jsonArray = parser.parse(jsonContent).asInstanceOf[JSONArray]
+      jsonArray.toArray.flatMap { obj =>
+        val jsonObject = obj.asInstanceOf[JSONObject]
+        Option(jsonObject.get("event-id").asInstanceOf[Long].toInt).map { eventId =>
+          val title = Option(jsonObject.get("title")).map(_.toString).getOrElse("")
+          val fullText = Option(jsonObject.get("fullText")).map(_.toString).getOrElse("")
+          val date = Option(jsonObject.get("date")).map(_.toString).getOrElse("")
+          JsonObject(eventId, date, title, fullText)
+        }
+      }.toList.sortBy(_.eventId)(Ordering[Int].reverse)
+    } catch {
+      case e: ParseException =>
+        throw new Exception(s"Error parsing JSON: ${e.getMessage}", e)
```

Comment:
For messages that might be of importance to the user and/or will be seen by the user, don't throw an exception or people will be annoyed and report it as a bug. In these cases, show a nice dialog, for example with `org.nlogo.swing.OptionDialog`.

Reply:
Done